NitroxServer/GameLogic/Entities/EntityData.cs

```diff
@@ -37,7 +37,7 @@ public class EntityData
         private Dictionary<string, Entity> entitiesByGuid = new Dictionary<string, Entity>();
 
         [ProtoIgnore]
-        private Dictionary<string, Entity> globalRootEntitiesByGuid = new Dictionary<string, Entity>();
+        public Dictionary<string, Entity> GlobalRootEntitiesByGuid = new Dictionary<string, Entity>();
```

Comment:
Instead of exposing the Dictionary I recommend creating a remove method `public bool Remove(GUID guid)` on EntityData which will lock the dictionary and try to remove said item returning true if successful.